scrapper.py

```diff
@@ -2,62 +2,232 @@
     """
+    pass
 
 
 class IncorrectNumberOfArticlesError(Exception):
     """
     Total number of articles to parse in not integer
     """
+    pass
 
 
 class Crawler:
     """
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        main_block_bs = article_bs.find('div', {'class': 'two_thirds'})
+        urls_bs = main_block_bs.find_all('a')
+        return urls_bs
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        articles_to_find = self.max_articles
+
+        for seed_url in self.seed_urls:
+            response = requests.get(url=seed_url, headers=HEADERS)
+            if not response.ok:
+                print('request failed')
+
+            soup = BeautifulSoup(response.text, 'lxml')
+            urls_bs = self._extract_url(soup)
+
+            for url_bs in urls_bs:
+                if articles_to_find == 0:
+                    break
+
+                if 'http://journals.tsu.ru' + url_bs['href'] not in self.urls:
+                    self.urls.append('http://journals.tsu.ru' + url_bs['href'])
+                    articles_to_find -= 1
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, full_url, i):
+        self.article_url = full_url
+        self.article_id = i
+        self.article = Article(self.article_url, self.article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        pdf_link = article_bs.find('a', {'class': 'file pdf'})['href']
+        article_instance = pdf_utils.PDFRawFile(pdf_link, self.article_id)
+        article_instance.download()
+        pdf_text = article_instance.get_text()
+        main_text = pdf_text.split('Литература \n')[0]
+        article_text = ''.join(main_text)
+        self.article.text = article_text
+
+    def _fill_article_with_meta_information(self, article_bs):
+        self.article.title = self._find_article_title(article_bs)
+        self.article.date = datetime.datetime.strptime(self._find_article_date(article_bs),
+                                                       "%Y-%m-%d")
+        self.article.author = self._find_article_author(article_bs)
+        self.article.topics = self._find_article_topics(article_bs)
+
+    def _find_article_author(self, article_bs):
+        main_body_bs = article_bs.find('div', {'class': 'two_thirds'})
+        authors_table_bs = main_body_bs.find('table', {'class': 'content_table otvet_list'})
+        authors_table_raws_bs = authors_table_bs.find_all('tr')[1:]
+        authors_list = []
+        for table_raw in authors_table_raws_bs:
+            authors_list.append(table_raw.find('td').text)
+        return ', '.join(authors_list)
```

Comment:
According to docs, in case there are several authors, you are expected to store their names as a list of strings.

Reply:
Done